src/rabbit_amqqueue_process.erl

```diff
@@ -1454,10 +1454,10 @@ handle_info({bump_credit, Msg}, State = #q{backing_queue       = BQ,
     %% rabbit_variable_queue:msg_store_write/4.
     credit_flow:handle_bump_msg(Msg),
     noreply(State#q{backing_queue_state = BQ:resume(BQS)});
-handle_info(bump_reduce_memory_use, State = #q{backing_queue       = BQ,
-                                               backing_queue_state = BQS}) ->
-    put(waiting_bump, false),
-    noreply(State#q{backing_queue_state = BQ:resume(BQS)});
+handle_info(bump_reduce_memory_use, State = #q{ backing_queue       = BQ,
+                                                backing_queue_state = BQS0 }) ->
+    BQS1 = BQ:handled_bump_reduce_memory_use(BQS0),
+    noreply(State#q{ backing_queue_state = BQ:resume(BQS1) });
```

Comment:
This code calls two callbacks, one of them just updates the state. Should it be a single callback instead?